### PR TITLE
fixed the issue with filters being taken off from rendering values

### PR DESCRIPTION
**Pull request by fonsekaean**

### CODE REVIEW

File: src/TagField.php [hunk 1]
```diff
@@ -275,7 +275,7 @@ protected function getOptions($onlySelected = false)
 
         if ($values) {
             if (is_array($values)) {
-                $values = DataList::create($dataClass)->filter($titleField, $values);
+                $values = $source->filter($titleField, $values); // DataList::create($dataClass)->filter($titleField, $values);
```
Comment (robbieaverill) [inline src/TagField.php]: Seems sensible. Would you mind removing the commented out code here, and updating the phpcs violation on line 393?

Comment (fonsekaean) [reply src/TagField.php]: @robbieaverill fixed those.

File: src/TagField.php [hunk 2]
```diff
@@ -390,6 +390,9 @@ protected function getOrCreateTag($term)
             $record = Injector::inst()->create($dataClass);
             $record->{$titleField} = $term;
             $record->write();
+            if ($source instanceof SS_List) {
+                $source->add($record);
+            }
```
Comment (robbieaverill) [inline src/TagField.php]: Would you mind explaining what this change does? As far as I can tell it's not used, since `$record` is what is returned

Comment (fonsekaean) [reply src/TagField.php]: This adds the new tag created to the Many_Many / Has_many list. So it completes the relation.

Comment (robbieaverill) [reply src/TagField.php]: Oh of course, because `->add()` internally writes the `$record`

**Comment by robbieaverill** [comment]
Thanks for the patch!

**Pull request merged**